scanpy/readwrite.py

```diff
@@ -351,15 +357,17 @@ def read_visium(
                 else:
                     raise OSError(f"Could not find '{f}'")
 
-        adata.uns['images'] = dict()
+        adata.uns["spatial"][library_id]['images'] = dict()
         for res in ['hires', 'lowres']:
             try:
-                adata.uns['images'][res] = imread(str(files[f'{res}_image']))
+                adata.uns["spatial"][library_id]['images'][res] = imread(
+                    str(files[f'{res}_image'])
+                )
             except Exception:
                 pass
 
         # read json scalefactors
-        adata.uns['scalefactors'] = json.loads(
+        adata.uns["spatial"][library_id]['scalefactors'] = json.loads(
             files['scalefactors_json_file'].read_bytes()
         )
 
```

Comment:
```suggestion
    adata.uns["spatial"][library_id]["metadata"] = {
        k: attrs[k] for k in ("chemistry_description", "software_version")
    }

```

For adding some metadata